cypress/support/pageObjects/performancePageLocators.js

```diff
@@ -0,0 +1,335 @@
+        const forceReload = true;
+        Cypress.config('defaultCommandTimeout', 4000);
+    
+        // Function to extract page name from URL
+        const extractPageName = (url) => {
+            const pageName = url.split('/').filter(Boolean).pop();
+            cy.log('Extracted page name:', pageName);
+            expect(pageName).to.not.be.empty;
+            return pageName;
+        };
+    
+        // Function to visit the site and check request count
+        const visitSiteAndCheckRequestCount = (url, pageName) => {
+            cy.get(this._excludeKeywordInputField)
+                .clear()
+                .type(pageName);
+    
+            cy.intercept('GET', url).as('apiRequest');
+            cy.get(this._visitSiteButton)
+                .invoke('removeAttr', 'target')
+                .click({ force: true });
+    
+            cy.reload(forceReload);
+            cy.get(this._samplePageButton).click();
+            cy.wrap(requestCount).should('equal', 0);
+    
+            cy.go('back').then(() => {
+                cy.go('back');
+            });
+        };
+    
+        // Function for dropdown interaction logic
+        const handleDropdownSelection = () => {
+            cy.get(this._dropDownForLinkPrefetch).then(($buttonLabel) => {
+                const selectedText = $buttonLabel.text().trim();
+    
+                if (selectedText === selectedDropDown) {
+                    cy.log('Second option is already selected. Proceeding with the test...');
+                    cy.get(this._dropDownForLinkPrefetch).should('have.text', selectedDropDown);
+                } else {
+                    cy.log('Second option is not selected. Selecting the second option...');
+                    cy.get(this._dropDownForLinkPrefetch).click();
+                    cy.get(this._mouseDownElement).click();
+                    cy.get(this._dropDownForLinkPrefetch).should('have.text', selectedDropDown);
+                }
+    
+                // Visit site first to make the sample page link visible
+                cy.get(this._visitSiteButton)
+                    .invoke('removeAttr', 'target')
+                    .click();
+    
+                // Wait for the sample page link to appear and extract its URL
+                cy.get('.wp-block-pages-list__item__link.wp-block-navigation-item__content', { timeout: 6000 })
+                    .should('be.visible')
+                    .invoke('attr', 'href')
+                    .then((url) => {
+                        const pageName = extractPageName(url);
+                        cy.go('back'); // Go back after extracting the URL
+    
+                        visitSiteAndCheckRequestCount(url, pageName);
+                    });
+            });
+        };
+    
+        handleDropdownSelection(); // Call the refactored function
+    }
+    interceptCallForMouseHoverWithoutExcludeRunTimeURL(selectedDropDown, statusCode) {
```

Comment:
Would it make sense to simplify all intercepts into three reusable functions? This will help us reuse the code better and also reduce repeated code. Maybe something like this:
```
  interceptRequest(method, url, alias) {
    cy.intercept(method, url).as(alias);
  }

  visitAndIntercept(url, alias) {
    this.visitSiteButton.invoke('removeAttr', 'target').click();
    cy.intercept('GET', url).as(alias);
    cy.reload(true);
  }

  assertApiRequest(alias, expectedStatusCode) {
    cy.wait(`@${alias}`).its('response.statusCode').should('eq', expectedStatusCode);
  }
```